modyn/config/schema/pipeline/trigger/cost/cost.py

```diff
@@ -0,0 +1,74 @@
+"""Cost aware triggers are evaluate the trade-off between the cost of a trigger
+(e.g. measured by training machine time) and the benefit of the trigger (e.g.
+measured by a regret metric like data incorporation latency or avoidable
+misclassification incorporation latency)."""
+
+from typing import Literal
+
+from pydantic import Field
+
+from modyn.config.schema.base_model import ModynBaseModel
+
+from ..performance.performance import _InternalPerformanceTriggerConfig
+
+
+class _CostTriggerConfig(ModynBaseModel):
+    """Base class for cost aware trigger policies."""
+
+    cost_tracking_window_size: int = Field(
+        description="How many trigger into the past should be considered for the linear cost w.r.t. data amount model."
+    )
+
+
+class DataIncorporationLatencyCostTriggerConfig(_CostTriggerConfig):
+    """Cost aware trigger policy configuration that uses the data incorporation
+    latency as a regret metric.
```

Comment:
I assume this latency describes how many seconds/minutes/hours a data has not been trained on, right? Do you base this off the timestamp of the data point (which could be in 1970) or based on the current timestamp at the supervisor?

Reply:
It's based on the timestamps of the batches (first and last sample) at the policy. The supervisor time is not used at all.

However, your comment made me think of the sample timestamps. Currently, I just assumed that if, e.g., the duration of the last batch is 10s and there were 50 samples, then 50*10s of latency will be added for the last interval (+ the additional latency for all data before the last interval). However, we could use the fine-grain time resolution of every sample to, e.g., let samples at the end of the batch have a smaller contribution than the earlier ones. It's a bit more painful to implement this but probably desireable, wdyt? If you agree, I'll adjust the other PR in that regard.

Reply:
I guess I am not fully sure yet what you mean by "duration of a batch". Is that max timestamp - min timestamp in the batch? But I mean yeah each sample has its own latency, this is nothing on a batch level. I understand it as total latency = \sum_{all not yet trained samples} sample latency, and sample latency = max timestamp observerd so far - sample timestamp.  But I might be missing something in this interpretation

Reply:
with batch i refer to the data intervals established through the `detection_interval` config variable (and not the supervisor batches)

Reply:
my comment was more about the duration of a batch, but as discussed that is max timestamp - end timestamp of last batch.